### ISSUE TITLE
conflict between  markdown.extensions.footnotes and markdown.extensions.nl2br or escapeall 

### ISSUE THREAD

**Issue by vanleo2001** [facelessuser/MarkdownPreview#113]
When I use markdownpreview in sublime text3, I find the conflict between markdown.extensions.nl2br and markdown.extensions.footnotes.

I want to allow single newline to register as a hardbreak, so I enable the python-markdown extension nl2br. I also want to use ython-markdown extension footnotes. But if enbale both extensions, markdownpreview will not render footnotes correctly. 

I trried python-markdown extension escapeall , it has the same error as nl2br.

picture1: both enabling extension footnotes and nl2br, footnotes render error
![snap1070](https://user-images.githubusercontent.com/12027772/81296135-0a4bd680-90a4-11ea-9b0d-94cd079ef3c4.jpg)

picture2: disabling extension nl2br, footnotes render ok
![snap1069](https://user-images.githubusercontent.com/12027772/81296177-159f0200-90a4-11ea-9e64-d467ed0dde53.jpg)

**Comment by facelessuser** [comment]
If there is an issue, the issue would either be in Python Markdown or pymdown-extensions, with the latter being the more likely case, with that said, it is probably more likely a user error, but I cannot determine without the actual Markdown source, not just a picture of the final render.

Please post the Markdown source so that I can determine where this issue actually belongs.

@gir-bot remove S: triage
@gir-bot add S: more-info-needed

**Comment by vanleo2001** [comment]
I use sublime text 3 (3211) x86 in win7 64bit. The encoding of markdown file is utf-8.


picture:  simple markdown text for test
![snap1076](https://user-images.githubusercontent.com/12027772/81358253-909a0400-9108-11ea-82d0-9a430b0d3b79.png)

picture:  All the configure of markdownpreview is default. Rendering footnotes is OK
![snap1072](https://user-images.githubusercontent.com/12027772/81358284-ab6c7880-9108-11ea-9560-7a7f21fda784.jpg)


picture: After enabling python-markdown extension nl2br, Rendering footnotes error. 
![snap1074](https://user-images.githubusercontent.com/12027772/81358309-c17a3900-9108-11ea-8817-2cc8957a6c8a.jpg)

picture: Footnotes are rendered as "\<a\>" in html source 
![snap1075](https://user-images.githubusercontent.com/12027772/81358355-d656cc80-9108-11ea-9e67-e7d77fbfbf35.jpg)

**Comment by facelessuser** [comment]
Please post a copy pastable examples of the source and of your configuration.

**Comment by facelessuser** [comment]
I have tried to reproduce and have absolutely no issues.

![chrome_isVPwgydtG](https://user-images.githubusercontent.com/1055125/81417564-9db6f300-9108-11ea-933b-ca9c231994d9.png)

Based on your picture above, you are only enabling `markdown.extensions.nl2br` and **not** enabling any other extensions, and since footnotes and escapeall, are both missing from your extensions setting, neither are enabled.

In short, this is a user error. When defining your own extension setup, you need to specify **all** extensions that you wish to enable as it overrides the defaults.

@gir-bot remove S: more-info-needed
@gir-bot add T: support
